_posts/en/newsletters/2019-12-04-newsletter.md

```diff
@@ -0,0 +1,155 @@
+    recursive [covenants][topic covenants]---script conditions that
+    apply not just to a finite set of descendant transactions but which
+    will apply to an infinite set of descendant transactions.  For
+    example, a spender could ensure that their coins could only ever
+    again be sent to a set of three pubkeys---any payment to any other
+    pubkey would be forbidden.  O'Connor's concerns seemed to focus on
+    this odd behavior making it harder to model the semantics of Bitcoin
+    Script, something which O'Connor has previously worked on and which
+    is related to his continuing work on the [Simplicity][] scripting
+    language.
+
+    On IRC, Gregory Maxwell and Jeremy Rubin [discussed][irc ctv]
+    several aspects of CTV, especially focusing on making the proposed
+    opcode easier to use with advanced designs without making it harder
+    to use with the simple [congestion controlled transactions][] and
+    [payment pools][] already proposed.  They also discussed whether it
+    was really necessary to prevent people from creating recursive
+    covenants, with possible allusions in the conversation to a [2013
+    thread][coincovenants] started by Maxwell about awful ways misguided
+    people might use recursive covenants.
+
+- **Proposed watchtower BOLT:** Sergi Delgado Segura [posted][watchtower
+  protocol] to the Lightning-Dev mailing list a draft BOLT he and
+  Patrick McCorry have been working on.  [Watchtowers][topic
+  watchtowers] are services that broadcast penalty transactions on
+  behalf of LN nodes that may be offline, recovering any funds that
+  would otherwise have been lost due to an old channel state being
+  confirmed onchain.  The goal of the proposed specification is to allow
+  all LN implementations to interoperate with any watchtower rather than
+  there being a different watchtower implementation for every LN
+  implementation.
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo],
+[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
+[libsecp256k1][libsecp256k1 repo], [Bitcoin Improvement Proposals
+(BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*
+
+- [C-Lightning #3259][] add experimental support for payment secrets,
+  which are proposed in [BOLTs #643][] as part of enabling [multipath
+  payments][topic multipath payments].  The payment secret is generated
+  by the receiver and included in their [BOLT11][] invoice.  The spender
+  includes this secret in the part of their payment that's encrypted to
+  the receiver's key.  The receiver only accepts an incoming payment for
+  the invoice if it contains the secret, preventing any other nodes from
+  probing the receiver to see whether it's expecting additional payments
+  to a previously-used payment hash.
+
+- [C-Lightning #3268][] changes the default network from Bitcoin testnet
+  to Bitcoin mainnet.  It also adds a new configuration option,
```

Comment:
Hopefully this change fixes a long-standing issue of needing to manually delete the testnet dir to switch to mainnet, without too much breakage (some was reported initially), and allows testnet and mainnet nodes to run at the same time.